lisa/sut_orchestrator/hyperv/serial_console_logger.ps1

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+param (
+    [Parameter(Mandatory=$true)][string]$PipeName,
+    [Parameter(Mandatory=$true)][string]$OutFile,
+    [Parameter(Mandatory=$false)][int]$PipeReconnectMillis = 12000
+)
+
+Write-Host "PID: $PID"
+Write-Host "PipeName: $PipeName"
+Write-Host "OutFile: $OutFile"
+
+$oStream = New-Object System.IO.FileStream $OutFile, 'OpenOrCreate', 'Write', 'Read'
+$pClient = New-Object System.IO.Pipes.NamedPipeClientStream(".", $PipeName)
+$reconnect = $false
+
+try {
+    while (1) {
+        $timeout = 60000
+        if ($reconnect) {
+            $timeout = $PipeReconnectMillis
+        }
+
+        $pClient.Connect($timeout)
+        if (!$pClient.IsConnected) {
+            Write-Host "Failed to connect to pipe. Exiting..."
+            exit
+        }
+
+        Write-Host "Connected to pipe"
+        $pClient.CopyTo($oStream)
```

Comment:
Is it a block operation?

Reply:
Yes.

Reply:
if it's a block operation, the loop doesn't need a sleep. if there is some case it's not blocked, only sleep, if the loop time is shorter than 1 second. It makes sure minimal loss of log.

Reply:
Which sleep are you referring to? This loop doesn't have any sleep. If we get disconnected, we try to immediately connect again.

Reply:
Oh, sorry. I mixed with other code snippets.